### DISCUSSION TITLE
How to know when a user started following another user?

### DISCUSSION THREAD

**Discussion by trifenol** [bluesky-social/atproto#2829] [How to do it?]
I’m working on a research project and would like to know when a user started following another user. I looked through the documentation but couldn’t find anything related to this issue.

**Comment by mary-ext** [comment]
Follow records has a createdAt timestamp set by the client, you can use that however it'd be inaccurate (can be manipulated, system time can be behind)

Assuming manipulation isn't that big of a deal, that's a worthwhile approach. Otherwise the only option is for you to be consuming the firehose and noting down the time that you received that follow record, that's as accurate you can get.

**Reply by bnewbold** [reply]
You could also parse the record key (TID) and map that to a timestamp. That is also not guaranteed accurate (users have full control over their repositories), but is probably fine for most cases and you can filter out bogus values (future dates, far past). It is an issue in antagonistic scenarios like bot networks.
